conan/tools/cmake/toolchain.py

```diff
@@ -344,6 +344,8 @@ def _apple_sdk_name(self):
     def context(self):
         os_ = self._conanfile.settings.get_safe("os")
         host_architecture = self._get_architecture()
+        if host_architecture is None:
```

Comment:
Shouldn't at least the CMAKE_SYSTEM_NAME, CMAKE_SYSTEM_VERSION for iOS, tvOS, watchOS be defined? Is it possible to have those target systems without arch?

Reply:
we can try...

Reply:
Yes, it is possible. It allowed to simplify the code, please check again.